Main.py

```diff
@@ -2,431 +2,409 @@
-@asyncio.coroutine
-def on_ready():
-    print('Logged in as {0} ({1})'.format(bot.user.name, bot.user.id))
-
-@bot.command(pass_context=True)
-@commands.has_role("idoneam")
-@asyncio.coroutine
-async def update(self):
+async def on_ready():
+    logger.info(f'Logged in as {bot.user.name} ({bot.user.id})')
+
+
+@bot.command()
+@commands.has_role(DEVELOPER_ROLE)
+async def update(ctx):
     """
     Update the bot by pulling changes from the git repository
     """
     shell_output = subprocess.check_output("git pull", shell=True)
     status_message = shell_output.decode("unicode_escape")
-    await self.bot.say("`%s`" % status_message)
-    
-@bot.command(pass_context=True)
-@commands.has_role("idoneam")
-@asyncio.coroutine
-async def backup(self):
-    '''
+    await ctx.send(f"`{status_message}`")
+
+
+@bot.command()
+@commands.has_role(DEVELOPER_ROLE)
+async def backup(ctx):
+    """
     Send the current database file to the channel
-    '''
-    current_time = datetime.datetime.now(tz=pytz.timezone('America/New_York')).strftime('%Y%m%d-%H:%M')
-    backup_filename = 'MiniScores_%s.db' % current_time
-    await self.bot.send_file(self.message.channel, DB_PATH, filename=backup_filename, content='Backup')
-    
-@bot.command(pass_context=True)
-@commands.has_role("idoneam")
-@asyncio.coroutine
-async def restart(self):
-    '''
+    """
+    current_time = datetime.datetime.now(tz=pytz.timezone(CROSSWORD_TIMEZONE)).strftime('%Y%m%d-%H:%M')
+    backup_filename = f'MiniScores_{current_time}.db'
+    await ctx.send("Backup", file=discord.File(DB_PATH, filename=backup_filename))
+
+
+@bot.command()
+@commands.has_role(DEVELOPER_ROLE)
+async def restart(ctx):
+    """
     Restart the bot
-    '''
-    await self.bot.say("`Beep boop boop beep beep beep beep`")
+    """
+    await ctx.send("```Beep boop boop beep beep beep beep```")
     python = sys.executable
     os.execl(python, python, *sys.argv)
 
-@bot.command(pass_context=True)
-@commands.has_role("crosswords")
-@asyncio.coroutine
-async def addtime(self, time: str = None):
+
+def _update_avg(conn: sqlite3.Connection, member) -> Tuple[Tuple[Optional[int], Optional[int]],
+                                                           Tuple[Optional[int], Optional[int]]]:
+    c = conn.cursor()
+
+    times_list = c.execute('SELECT Score, Date FROM Scores WHERE ID = ?', (member.id,)).fetchall()
+    reg_vals = []
+    sat_vals = []
+    for score, score_date in times_list:
+        (sat_vals if _get_day_from_ymd(score_date) == "Sat" else reg_vals).append(int(score))
+
+    user_avgs = c.execute('SELECT RegAvg, SatAvg FROM Ranking WHERE ID = ?', (member.id,)).fetchone()
+    if not user_avgs:
+        user_avgs = (None, None)
+
+    old_reg_avg: Optional[int]
+    old_sat_avg: Optional[int]
+    old_reg_avg, old_sat_avg = user_avgs
+
+    new_reg_avg: Optional[int] = None
+    new_sat_avg: Optional[int] = None
+
+    # A regular average can only be set if regular crossword scores exist
```

Comment:
Average should be deleted if there's no score (if someone has one score and deletes it, the avg will stay the same because of this)

Reply:
Resolved in 20d3d11